src/main/groovy/nextflow/executor/OarExecutor.groovy

```diff
@@ -0,0 +1,159 @@
+
+    /**
+     * Gets the directives to submit the specified task to the cluster for execution
+     *
+     * @param task A {@link TaskRun} to be submitted
+     * @param result The {@link List} instance to which add the job directives
+     * @return A {@link List} containing all directive tokens and values.
+     */
+
+    protected List<String> getDirectives( TaskRun task, List<String> result ) {
+        assert result !=null
+
+        result << '-n' << getJobNameFor(task)
+        result << '-d' << quote(task.workDir)
+
+        // the requested queue name
+        if( task.config.queue ) {
+            result << '-q'  << (String)task.config.queue
+        }
+
+        // miscellaneous cluster options semicollon separated
+        if( task.config.clusterOptions ) {
+          for (String item : task.config.clusterOptions.toString().tokenize(';')) {
+            result << item << ''
+          }
+        }
+
+      return result
+
+    }
+
+    @Override
+    String getHeaders( TaskRun task ) {
+        String result = super.getHeaders(task)
+        result += "cd ${quote(task.workDir)}\n"
+        return result
+    }
+
+    @Override
+    String getJobNameFor( TaskRun task ) {
+        def result = super.getJobNameFor(task)
+        // some implementations do not allow parenthesis in the job name -- see #271
+        result = result.replace('(','').replace(')','')
+        // PBS does not allow more than 15 characters for the job name string
+        result && result.size()>15 ? result.substring(0,15) : result
+    }
+    /**
+     * The command line to submit this job
+     *
+     * @param task The {@link TaskRun} instance to submit for execution to the cluster
+     * @param scriptFile The file containing the job launcher script
+     * @return A list representing the submit command line
+     */
+    List<String> getSubmitCommandLine(TaskRun task, Path scriptFile ) {
+        scriptFile.setPermissions(7,0,0)
+        return [ "oarsub", "-S", "./${scriptFile.getName()}"]
+    }
+
+    protected String getHeaderToken() { '#OAR' }
+
+    /**
+     * Parse the string returned by the {@code qsub} command and extract the job ID string
+     *
+     * @param text The string returned when submitting the job
+     * @return The actual job ID string
+     */
+
+    @Override
+    def parseJobId(String text) {
+    def pattern = ~ /OAR_JOB_ID=(\d+)/
+    for( String line : text.readLines() ) {
+        def m = pattern.matcher(line)
+        if( m.matches() ) {
+            return m.group(1).toString()
+            }
+        }
+
+        throw new IllegalStateException("Invalid OAR submit response:\n$text\n\n")
+    }
+
+    @Override
+    protected List<String> getKillCommand() { ['oardel'] }
+
+    @Override
+    protected List<String> queueStatusCommand(Object queue) {
+        String cmd = 'oarstat -f'
+        if( queue ) cmd += ' ' + queue
+        return ['sh','-c', "$cmd | egrep '(Job_Id:|state =)' ".toString()]
```

Comment:
I have no access to OAR cluster. Have you tested this command line? why the grep is needed?